pkg/utils/flow_utils.go

```diff
@@ -141,29 +147,24 @@ func AddTcpFlags(f *flow.Flow, syn, ack, fin, rst, psh, urg uint16) {
 	}
 }
 
-// Add TSval/TSecr to the flow as TCP ID.
+// Add TSval/TSecr to the flow's metadata as TCP ID.
 // The TSval/TSecr works as ID for the flow.
 // We will use this ID to calculate latency.
-func AddTcpID(f *flow.Flow, id uint64) {
-	if f.L4 == nil || f.L4.GetTCP() == nil {
-		return
-	}
-	k := &RetinaMetadata{}      //nolint:typecheck
-	f.Extensions.UnmarshalTo(k) //nolint:errcheck
-	k.TcpId = id
-	f.Extensions, _ = anypb.New(k) //nolint:errcheck
+func (meta *RetinaMetadata) AddTCPID(id uint64) {
```

Comment:
it's probably not a good idea to attach methods to a generated struct like this. do we actually need them? they just look like setters on already public fields

Reply:
yeah, i was debating between this and `AddTCPID(meta *RetinaMetadata, id unint64)`, seems like we have a consensus with @anubhabMajumdar comment 🙂